src/domains/billing/Checkout/components/CheckoutPage.tsx

```diff
@@ -0,0 +1,99 @@
+"use client";
+
+import { Elements } from "@stripe/react-stripe-js";
+import { Stripe, loadStripe } from "@stripe/stripe-js";
+import { Card, Layout, Text, TextContainer } from "@shopify/polaris";
+
+import { PageWithNotifications } from "src/components";
+import { CheckoutForm } from "./CheckoutForm";
+import { useCheckoutTaskFetch } from "../hooks";
+
+let stripePromise: Promise<Stripe | null>;
+const getStripe = () => {
+  if (!stripePromise) {
+    stripePromise = loadStripe(process.env.NEXT_PUBLIC_STRIPE_PK!);
+  }
+  return stripePromise;
+};
+
+/**
+ * Returns an error message based on the payment status.
+ * @param paymentStatus
+ * @returns an error message
+ * @error Throws error when a payment status is either "payment_pending" or "payment_due"
+ */
+export function getErrorMessageFromPaymentStatus(paymentStatus?: string) {
+  switch (paymentStatus) {
+    case "paid":
+      return "Checkout Error: The Task has been paid";
+    case "pay_in_processing":
+      return "Checkout Error: The payment is currently is being processed";
+    case undefined:
+      return "Checkout Error: The Task has been not completed yet";
+    default:
+      throw new Error(
+        "A valid paymentStatus was passed into getErrorMessageFromPaymentStatus"
+      );
+  }
+}
+
+export function isPaymentDue(paymentStatus?: string) {
+  return (
+    paymentStatus === "payment_due" || paymentStatus === "payment_declined"
+  );
+}
```

Comment:
Didn't write tests for these two functions because I don't think they are high value.

If someone wants to practice testing, happy to give this one out as an "easy-win" ticket.